pylabrobot/resources/ml_star/troughs.py

```diff
@@ -0,0 +1,63 @@
+""" Definitions for Hamilton-manufactured Troughs """
+
+# pylint: disable=empty-docstring
+# pylint: disable=invalid-name
+# pylint: disable=line-too-long
+
+from pylabrobot.resources.trough import (
+  Trough,
+  TroughBottomType
+)
+
+# # # # # # # # # # Hamilton_1_trough_200ml_Vb # # # # # # # # # #
+
+
+def _compute_volume_from_height_Hamilton_1_trough_200ml_Vb(h: float):
+  """ Function to compute volume of liquid in trough,
+  based on poylonmial fit of z-probed, empirical data.
+  """
+  if h > 89: # Maximal measured height possible
+    raise ValueError(f"Height {h} is too large for Hamilton_1_trough_200ml_Vb")
+  a, b, c, d = 5.50216554e-02, -5.30245236, 3.61064575e+03, -1.56155485e+04
+  polynomial_fit_of_empirical_data = a * h**3 + b * h**2 + c * h + d
```

Comment:
why do you use a 3d polynomial?

linear or maybe a parabola seems better visually, based on your data in the PR text:

<img width="592" alt="image" src="https://github.com/PyLabRobot/pylabrobot/assets/36956818/c2fdbd74-9e28-4a8d-bdb5-3070e060b68b">

Reply:
I compared linear and polynomial fitting of the measurements to make this decision:

![image](https://github.com/PyLabRobot/pylabrobot/assets/122165124/4ed8b459-3427-4839-8e9b-b40393a589fc)

These functions need be really precise, so I opted for what yielded the best RMSE values. I can see a point to be made about the precision maybe being too dependent on my specific setup, no matter how thoroughly I went through the testing to ensure accuracy. But of course, a polynomial fit will always be more precise, and since the extra computational load is insignificant I didn't see a reason why to simplify it.

It is quite a deviation from the geometry-first approach I have pursued previously but I found these fits to simply work better.
(I actually believe now that this is the reason for the unexplained functions we see in VENUS as well - though I hope we can document this type of information enough for users to understand what is happening).

Reply:
Tbh, this looks like overfitting to me. Why would it not be linear? Does it reproduce like this? Happy to merge and I think going empirically can make sense for complex geometry, I'm just skeptical :)

Reply:
It is non-linear because the cross-section is a rectangle with one half-circle attached to one end that partitions into two sections along the z-axis: the bottom is cone-shaped but not fully, instead it has a flat surface of about 2-3 mm with (kinda a frustum of a pyramid style), while the top continues the cross-section in a semi-linear way.
It is semi-linear because in the middle of the trough there is a partition wall, which is open above and below. This wall is designed to avoid the liquid inside the trough to take up momentum when the trough is moved, i.e. reduce splashing, but as a consequence it takes up volume of the trough overal, adding to the non-linear liquid filling behaviour.

I am not sure there is such a thing as over-fitting with getting height and volume calculations of liquids in containers right :)